common/protos/proto/user.proto

```diff
@@ -0,0 +1,186 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package databend_proto;
+
+import "datatype.proto";
+
+message AuthInfo {
+  uint64 ver = 100;
+
+  message None {}
+  message Password {
+    enum PasswordHashMethod {
+      PlainText = 0;
+      DoubleSha1 = 1;
+      Sha256 = 2;
+    }
+    bytes hash_value = 1;
+    PasswordHashMethod hash_method = 2;
+  }
+  message JWT {}
+
+  oneof info {
+    None none = 1;
+    Password password = 2;
+    JWT jwt = 3;
+  }
```

Comment:
What aboutt using a single `Empty` message for all kinds of enum variants that do not have a body?
Using `Empty` for `None` and `JWT` will get rid of several empty messages.

I.e., trade specificity for simplicity.